### ISSUE TITLE
[BUG]: Cannot connect to hardware encoder from Chrome, get only vp8

### ISSUE THREAD

**Issue by yvanzine** [Unity-Technologies/com.unity.webrtc#709]
### Package version

2.4.0-exp.7

### Environment

```markdown
Package version
2.4.0-exp.7

Environment
Linux, Ubuntu 20.04
Chromium 100.0.4896.127.
Unity Version 2020.3.24f1
Packages:
[com.unity.webrtc 2.4.0-exp.7]
[UnityRenderStreaming experimental/webrtc-2.4.0-exp.7 branch]
```

### Steps To Reproduce

I have been testing 2.4.0-exp.7 webrtc package to see if quality artifacts have been addressed. The main issue is that I can only connect from the browser with a vp8 stream. At 1920x1080 resolution software encoder gives only 8fps which is a non-starter for us. I believe web samples in UnityRenderStreaming will have to be updated to enable the 2.4.0-exp.7 support. I also suspect logic on the backend is also too restrictive and it falls back on vp8 too eagerly once it detects that h264 is available, it may be still choosing vp8. NOTE that h264 software decoder is still faster than vp8 because the backend is so slow in vp8. 

Hardware encoder at HD (1920x1080) or 4K (3840x2160) is the only way to go and that is why I am filing this bug, because there is a perception that vp8 is a valid default, but it is very much not.
 
Please change the default to be H264 as it was before.
 

### Current Behavior

vp8 is the default and it is wrong.

### Expected Behavior

Need H264 to be the default.

**Comment by karasusan** [comment]
@yvanzine 
Hi, we are testing Unity Render Streaming with H264 codec.
There are two branches on the repository. You can use branches which you prefer.
https://github.com/Unity-Technologies/UnityRenderStreaming/tree/experimental/webrtc-2.4.0-exp.7
https://github.com/Unity-Technologies/UnityRenderStreaming/tree/test/nvcodec

**Comment by yvanzine** [comment]
@karasusan I was using this branch:
https://github.com/Unity-Technologies/UnityRenderStreaming/tree/experimental/webrtc-2.4.0-exp.7

I was not able to connect with H264 from the client, only vp8. Should I be using the other branch?

**Comment by kannan-xiao4** [comment]
@yvanzine Could you try this one https://github.com/Unity-Technologies/UnityRenderStreaming/tree/test/nvcodec ?
I can connect with H264 codec in localhost.
However, I recommend a streaming resolution of 1280x720. (In FullHD, there is a problem with https://github.com/Unity-Technologies/com.unity.webrtc/issues/702)

**Comment by yvanzine** [comment]
@matt-knowles-level-ex  your suggestion unfortunately did not work
@kannan-xiao4 
@karasusan 

I have tried both branches and I am still getting only vp8 software encoder to connect.

It is strange but bool hardwareEncoder parameter is completely ignored by RenderStreaming, RenderStreamingDependencies and RenderStreamingInternal class. There is no code that does anything with the hardwareEncoder. Maybe it is always true if not set. But there is just nothing passed from the UI and the checkbox Hardware Encoder Support does not do anything.

https://github.com/Unity-Technologies/UnityRenderStreaming/blob/test/nvcodec/com.unity.renderstreaming/Runtime/Scripts/RenderStreaming.cs

By the way, I have been using WebBrowserInput.unity, WebBrowserInputSample.cs and modified web app to test. 

@kannan-xiao4 Can you please provide detailed description of what you do to test hardware encoder on your system?

**Comment by karasusan** [comment]
@yvanzine
I am not testing WebBrowserInput sample. How about the Broadcast sample?

**Comment by yvanzine** [comment]
@karasusan Can you please provide more specific instructions for Broadcast sample? E.g. what you do on the client and inside the scene, so that I do not make a mistake. Point me to the doc that matches closest the procedure.

**Comment by karasusan** [comment]
@yvanzine 
https://docs.unity3d.com/Packages/com.unity.renderstreaming@3.1/manual/sample-broadcast.html

**Comment by karasusan** [comment]
@yvanzine
Hi, could you tell me your current situation?

**Comment by yvanzine** [comment]
@karasusan I will try the broadcast test tomorrow and get back to you.

**Comment by yvanzine** [comment]
I was able to get h264 hardware encoder to work with broadcast sample from the following branches:
com.unity.webrtc 2.4.0-exp.7
https://github.com/Unity-Technologies/UnityRenderStreaming/tree/test/nvcodec

https://user-images.githubusercontent.com/19309404/167722846-8c6014b7-284c-4256-9c0b-c7cbab6674be.mp4

**Comment by yvanzine** [comment]
@karasusan A few observations:

1) I cannot reproduce the smearing bug which is great.
 https://github.com/Unity-Technologies/com.unity.webrtc/issues/205
What was the root cause of the smearing?

2) I still can see fuzzy pulse (we know that this has not been fixed probably because the plugin is changing bitrate too frequently)
3) I see a crash at HD+ resolution which is probably due to incorrect encoder profile
4) The framerate is very uneven (probably due to too frequent bitrate adjustment) and there is loss of performance. We lost about 15 fps from 2.3.3-preview

**Comment by karasusan** [comment]
@yvanzine 
Thanks for sharing!

1. I assume that the cause of the issue is the load of rendering thread.
2. Already fixed in this PR #716.
3. Testing in this PR #721.
4. Please check again after merged them.
